### PR TITLE
Update ink! caller integration test

### PR DESCRIPTION
**Pull request by xermicus**
Updates the caller contract to use the latest ink! release. I also re-structured the solidity code to try and make what's going on in there obvious.

### CODE REVIEW

File: integration/substrate/inkee.sol [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+function ink_result(uint32 _value) returns (InkResult) {
+    return InkResult({ok: 0, value: _value});
+}
+
 contract Inkee {
+    // We use this selector just for convenience
     @selector([1, 2, 3, 4])
-    function echo(uint32 v) public pure returns (bool, uint32) {
-        return (false, v);
+    function echo(uint32 v) public returns (InkResult) {
```
Comment (seanyoung) [inline integration/substrate/inkee.sol:21]: why can't this function be pure any more? Is this a limitation of free standing functions?

Comment (seanyoung) [reply integration/substrate/inkee.sol:21]: I just couldn't help myself and tested it - if you declare `function ink_result()` as `pure` then `function echo()` can be pure too, if that matters at all.

File: integration/substrate/inkee.sol [hunk 2]
```diff
@@ -1,6 +1,24 @@
+// ink! contracts are expected to return a value of type `Result<T, LangError>` (rusts "Result" type).
```
Comment (LucasSte) [inline integration/substrate/inkee.sol:1]:
```suggestion
// ink! contracts are expected to return a value of type `Result<T, LangError>` (Rust's "Result" type).
```

File: integration/substrate/inkee.sol [hunk 3]
```diff
@@ -1,6 +1,24 @@
...
+// The ink! call builder only supports decoding the result into this `Result` type.
+// This works fine for ink! because ink! contracts are supposed to always return this type.
+// 
+// In solidity, nor does this type exist, neither does the concept of a "LangError".
```
Comment (LucasSte) [inline integration/substrate/inkee.sol:5]:
```suggestion
// In Solidity, not only does this type not exist, but also the concept of a "LangError" is inexistent.
```

**Review by LucasSte — APPROVED** [review]

File: integration/substrate/inkee.sol (hunk 1 above)
Comment (xermicus) [reply integration/substrate/inkee.sol:21]: Good idea

**Pull request merged**